packages/dev/core/src/assetContainer.ts

```diff
@@ -438,7 +433,20 @@ export class AssetContainer extends AbstractScene {
                 onNewCreated(instancedNode, replicatedInstancedNode);
             } else {
                 // Mesh or TransformNode
-                const canInstance = !localOptions?.doNotInstantiate && (node as Mesh).getTotalVertices() > 0;
+                let canInstance = true;
+                if (
+                    node.getClassName() === "TransformNode" || 
+                    (node as Mesh).skeleton || 
+                    (node as Mesh).getTotalVertices() === 0) {
+                    // Transform nodes, skinned meshes, and meshes with no vertices can never be instanced!
+                    canInstance = false;
+                } else if (localOptions?.doNotInstantiate) {
+                    if (localOptions.doNotInstantiate instanceof Function) {

```

Comment:
We generally use `typeof XXX === "function"` in the code base (I think it's more performant?).